code/__HELPERS/unsorted.dm

```diff
@@ -1517,3 +1517,20 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 	for(var/V in GLOB.player_list)
 		if(is_servant_of_ratvar(V) || isobserver(V))
 			. += V
+
+//datum may be null, but it does need to be a typed var
+#define NAMEOF(datum, X) (list(##datum.##X, #X)[2])
+
+#define VARSET_LIST_CALLBACK(target, var_name, var_value) CALLBACK(GLOBAL_PROC, /proc/___callbackvarset, ##target, ##var_name, ##var_value)
+//dupe code because dm can't handle 3 level deep macros
+#define VARSET_CALLBACK(datum, var, var_value) CALLBACK(GLOBAL_PROC, /proc/___callbackvarset, ##datum, NAMEOF(##src, ##var), ##var_value)
```

Comment:
where are you getting `##src`?

Reply:
read the ref dunc

Reply:
I'm familiar with `##param` in macros but `src` is not a parameter in this define so I'm not sure what this is doing

Reply:
oh well that's just cyberboss being dumb.

Reply:
Probably he forgot to change it after switching names in the macro. src used to be the name of one of the args.